tornado/ioloop.py

```diff
@@ -625,6 +632,35 @@ def add_future(self, future, callback):
         future.add_done_callback(
             lambda future: self.add_callback(callback, future))
 
+    def run_in_executor(self, executor, func, *args):
+        """Runs a function in a ``concurrent.futures.Executor``. If
+        ``executor`` is ``None``, the IO loop's default executor will be used.
+
+        Use `functools.partial` to pass keyword arguments to `func`.
+
+        """
+        if not _concurrent_futures_found:
+            raise RuntimeError(
+                "concurrent.futures is required to use IOLoop.run_in_executor")
+
+        if executor is None:
+            if not hasattr(self, '_executor'):
+                self._executor = ThreadPoolExecutor(max_workers=(cpu_count() * 5))
+            executor = self._executor
+
+        future = executor.submit(func, *args)
+        self.add_future(future, lambda future: future.result())
```

Comment:
We shouldn't call `add_future` here. That will log the future's result even if the caller handles it.